draft-ietf-quic-transport.md

```diff
@@ -7411,6 +7402,43 @@ The initial contents of this registry are shown in {{iana-error-table}}.
 
 --- back
 
+# Sample Packet Number Encoding Algorithm {#sample-packet-number-encoding}
+
+The pseudo-code in {{alg-encode-pn}} shows how an implementation can select
+an appropriate size for packet number encodings.
+
+The EncodePacketNumber function takes two arguments:
+
+* full_pn is the full packet number of the packet being sent.
+* largest_acked is the largest packet number which has been acknowledged by the
+  recipient in the current packet number space, if any
+
+~~~
+EncodePacketNumber(full_pn, largest_acked):
+  // If no packets in the current space have been
+  // acknowledged, the full packet number is used.
+  if largest_acked is None:
+    return full_pn
+
+  // The number of bits must be at least one more
+  // than the base-2 logarithm of the number of contiguous
+  // unacknowledged packet numbers, including the new packet.
+  num_unacked = full_pn - largest_acked
+  min_bits = log(num_unacked, 2) + 1
+  num_bytes = ceil(min_bits / 8)
+  return full_pn[-num_bytes:]
```

Comment:
Someone has been doing too much python.

As the value is not encoded into bytes before you index it, this is hard to follow.

```suggestion
  let encoded = []
  for i in 0..num_bytes:
    encoded.unshift((full_pn >> (8 * i)) & 0xff)
    // encoded.unshift((full_pn / (256^i)) % 256) ?
  return encoded
```

Reply:
Someone has been doing too much Javascript :-)
I don't think `.unshift` can be used without definition like this (I had to look it up).

This will be tricky to do without language conventions. I would recommend simply:
```
// return the value after truncating the full packet
// number to num_bytes least significant bytes
return truncate(full_pn, num_bytes)

Reply:
So do it this way instead?

```
    encoded = [(either expression), encoded...]
```

Reply:
I like the solution of throwing it to an undefined function with a comment describing what it does.  The implementer can do that however their language best permits.